src/containers/Author.js

```diff
@@ -24,13 +25,22 @@ const authorDefaultImg = '/asset/author-default-img.svg'
 class Author extends React.Component {
   static fetchData({ params, store }) {
     const authorId = _.get(params, 'authorId', '')
-    return store.dispatch(fetchAuthorCollectionIfNeeded(authorId))
+    return Promise.all([ store.dispatch(fetchAuthorCollectionIfNeeded(authorId)), store.dispatch(fetchAuthorDetails(authorId)) ])
+      .catch(error => {
+        console.error(error) // eslint-disable-line no-console
+      })
   }
   componentDidMount() {
-    const authorId = this.props.params['authorId']
-    let { articlesByAuthor, fetchAuthorCollectionIfNeeded } = this.props
-    if (!articlesByAuthor[authorId]) {
-      fetchAuthorCollectionIfNeeded(authorId)
+    const { articlesByAuthor } = this.props
+    const authorId = _.get(this.props, 'params.authorId')
+    if (authorId) {
+      if (!articlesByAuthor[authorId]) {
+        this.props.fetchAuthorCollectionIfNeeded(authorId)
+      }
```

Comment:
`fetchAuthorCollectionIfNeeded`  would check the condition `articlesByAuthor[authorId]`